### PR TITLE
Audio Recoder and other minor fix.

### PR DESCRIPTION
**Pull request by poormusic2001**
1. Add Audio Recoder and its Simple UI.
2. Add file-attatchment.imageset back.

### CODE REVIEW

File: Monal/Classes/MLAudioRecoderManager.m [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+-(void)start{
+    
+    NSError *audioSessionCategoryError = nil;
+    NSError *audioRecodSetActiveError = nil;
+    AVAudioSession *audioSession = [AVAudioSession sharedInstance];
+    [audioSession setCategory:AVAudioSessionCategoryRecord error:&audioSessionCategoryError];
+    [audioSession setActive:YES error:&audioRecodSetActiveError];
+    if (audioSessionCategoryError) {
+        DDLogError(@"Audio Recoder set category error: %@", audioSessionCategoryError);
```
Comment (FriedrichAltheide) [inline Monal/Classes/MLAudioRecoderManager.m:34]: show warning to user? self.recoderManagerDelegate notifyResult:NO

File: Monal/Classes/MLAudioRecoderManager.m [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    if (audioRecodSetActiveError) {
+        DDLogError(@"Audio Recoder set active error: %@", audioRecodSetActiveError);
+    }
+    
+    NSError *recoderError = nil;
+    NSDictionary *recodSettings = [NSDictionary dictionaryWithObjectsAndKeys:
+                                   [NSNumber numberWithInt:kAudioFormatMPEG4AAC] , AVFormatIDKey,
+                                   [NSNumber numberWithInt:AVAudioQualityMin],AVEncoderAudioQualityKey,
+                                   [NSNumber numberWithInt: 1], AVNumberOfChannelsKey,
+                                   [NSNumber numberWithFloat:32000.0], AVSampleRateKey, nil];
```
Comment (FriedrichAltheide) [inline Monal/Classes/MLAudioRecoderManager.m:50]: @tmolitor-stud-tu what do you think about the sample rate?

File: Monal/Classes/MLAudioRecoderManager.m [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    [self.recoderManagerDelegate notifyStop:audioFileURL];
+}
+
+-(void)updateTimeInfo{
+    [self.recoderManagerDelegate updateCurrentTime:self.audioRecorder.currentTime];
+}
+
+- (void)audioRecorderDidFinishRecording:(AVAudioRecorder *)recorder successfully:(BOOL)flag{
+    if (flag) {
+        [self.recoderManagerDelegate notifyResult:YES error:@""];
```
Comment (FriedrichAltheide) [inline Monal/Classes/MLAudioRecoderManager.m]: can't you send "error:nil"?

File: Monal/Classes/chatViewController.m [hunk 4]
```diff
@@ -181,13 +186,19 @@ -(void) viewDidLoad
     [self setChatInputHeightConstraints:YES];
     
     if (@available(iOS 13.0, *)) {
-        [self.sendButton setImage:[UIImage systemImageNamed:@"paperplane.fill"] forState:UIControlStateNormal];
+        [self.sendButton setImage:[UIImage systemImageNamed:@"mic"] forState:UIControlStateNormal];
```
Comment (FriedrichAltheide) [inline Monal/Classes/chatViewController.m]: I would prefer a separate button for the mic action that would only be visible if `chatInput.length == 0`

**Review by FriedrichAltheide — CHANGES_REQUESTED** [review]

**Comment by Echolon** [comment]
> 1. Add Audio Recoder and its Simple UI.
> 2. Add file-attatchment.imageset back.

Does this close #238 ?

**Comment by poormusic2001** [comment]
@FriedrichAltheide 

Add another commit. 
I modify the the empty string to nil and also add another button for record button.

File: Monal/Classes/chatViewController.m [hunk 5]
```diff
@@ -87,6 +87,11 @@ @interface chatViewController()<IDMPhotoBrowserDelegate, ChatInputActionDelegage
 @property (nonatomic, strong) void (^editingCallback)(NSString* newBody);
 @property (nonatomic, strong) NSMutableSet* previewedIds;
 
+@property (atomic) BOOL isAudioMessage;
```
Comment (FriedrichAltheide) [inline Monal/Classes/chatViewController.m:90]: you never seem to be reading `isAudioMessage`

File: Monal/Classes/chatViewController.m [hunk 6]
```diff
@@ -927,6 +950,30 @@ -(IBAction)sendMessageText:(id)sender
...
+            if (granted) {
+                if (gestureRecognizer.state == UIGestureRecognizerStateBegan){
+                    [[MLAudioRecoderManager sharedInstance] setRecoderManagerDelegate:self];
+                    [[MLAudioRecoderManager sharedInstance] start];
+                } else if (gestureRecognizer.state == UIGestureRecognizerStateEnded){
+                    [[MLAudioRecoderManager sharedInstance] stop];
+                }
+            } else {
+                dispatch_async(dispatch_get_main_queue(), ^{
+                    UIAlertController *messageAlert =[UIAlertController alertControllerWithTitle:NSLocalizedString(@"Please Allow Audio Access",@ "") message:NSLocalizedString(@"If you want to use audio message you will need to allow access in Settings-> Privacy-> Microphone.",@ "") preferredStyle:UIAlertControllerStyleAlert];
```
Comment (FriedrichAltheide) [inline Monal/Classes/chatViewController.m:964]: `, @"")`

**Review by FriedrichAltheide — COMMENTED** [review]
@Echolon #238 should be kept open. The inline Audio UI needs a bit more love...

**Pull request merged**